flex-checks/src/main/java/org/sonar/flex/checks/OneStatementPerLineCheck.java

```diff
@@ -37,19 +41,44 @@
   tags = Tags.CONVENTION)
 @ActivatedByDefault
 @SqaleConstantRemediation("1min")
-public class OneStatementPerLineCheck extends AbstractOneStatementPerLineCheck<LexerlessGrammar> {
+public class OneStatementPerLineCheck extends FlexCheck {
+
+  private final Map<Integer, Integer> statementsPerLine = Maps.newHashMap();
+
+  @Override
+  public List<AstNodeType> subscribedTo() {
+    return Arrays.asList(FlexGrammar.STATEMENT, FlexGrammar.VARIABLE_DECLARATION_STATEMENT);
+  }
 
   @Override
-  public void init() {
-    subscribeTo(FlexGrammar.STATEMENT, FlexGrammar.VARIABLE_DECLARATION_STATEMENT);
+  public void visitFile(AstNode astNode) {
+    statementsPerLine.clear();
   }
 
   @Override
-  public AstNodeType getStatementRule() {
-    return null;
+  public void visitNode(AstNode statementNode) {
+    if (!isExcluded(statementNode)) {
+      int line = statementNode.getTokenLine();
+
+      if (!statementsPerLine.containsKey(line)) {
```

Comment:
I wonder if you could not replace this 4 lines of code simply by:

```
statementsPerLine.compute(line, (k, v) -> v == null ? 0 : v + 1);
```

Reply:
Done: 20cfed8